File: code/backend/java/core/src/main/java/games/paths/core/model/story/StorySummary.java
package games.paths.core.model.story;

/**
 * StorySummary - Domain model for a lightweight story listing entry.
 * Contains only the fields needed to display a story in a catalogue/list view.
 * Immutable record.
 */
public record StorySummary(
        String uuid,
        String title,
        String description,
        String author,
        String category,
        String group,
        String visibility,
        int priority,
        int peghi,
        int difficultyCount,
        CardInfo card) {

    public StorySummary {
        if (uuid == null || uuid.isBlank()) {
            throw new IllegalStateException("uuid is required");
        }
    }
}

File: code/backend/php/src/Core/Domain/Story/StorySummary.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class StorySummary
{
    public function __construct(
        public string $uuid,
        public ?string $title = null,
        public ?string $description = null,
        public ?string $author = null,
        public ?string $category = null,
        public ?string $group = null,
        public ?string $visibility = null,
        public int $priority = 0,
        public int $peghi = 0,
        public int $difficultyCount = 0
    ) {
    }
}
